kernel/src/table_properties/deserialize.rs

```diff
@@ -76,17 +76,33 @@ fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
         }
         "delta.checkpointPolicy" => props.checkpoint_policy = CheckpointPolicy::try_from(v).ok(),
         "delta.enableRowTracking" => props.enable_row_tracking = Some(parse_bool(v)?),
+        "delta.enableInCommitTimestamps" => {
+            props.enable_in_commit_timestamps = Some(parse_bool(v)?)
+        }
+        "delta.inCommitTimestampEnablementVersion" => {
+            props.in_commit_timestamp_enablement_version = Some(parse_uint(v)?)
```

Comment:
`parse_uint` already returns an option. Why not just:
```suggestion
            props.in_commit_timestamp_enablement_version = parse_uint(v)
```

Reply:
ah i actually had to think about this for a sec too lol - we actually want to return `None` in the case it doesn't parse. if we leave it as just `parse_uint(v)` then it will set the value to `None` and return `Some`

perhaps we should rethink this code and try to rewrite it more nicely considering both you and I have now stumbled on it :)

Reply:
Ah you're right! Thx for the clarification

Reply:
yep exactly. added a comment to clarify but given this has tripped up _three_ of us now I think we should rewrite lol

made a quick issue: https://github.com/delta-io/delta-kernel-rs/issues/682